text/0016-standard-modes.md

```diff
@@ -0,0 +1,357 @@
+        <param index="4" reserved="true" default="0"/>
+        <param index="5" reserved="true" default="0"/>
+        <param index="6" reserved="true" default="0"/>
+        <param index="7" reserved="true" default="NaN"/>
+      </entry>
+```
+
+
+## Getting Available Modes
+
+A new message will be added for enumerating all available modes (standard, base, and custom) for the current vehicle type:
+- The message will include the total number of modes and the index of the current mode.
+  This indexing is provided to allow a GCS to confirm that all modes have been collector and re-request any that are missing.
+  It is internal and should not be relied upon to have any order between reboots.
+- If there is a direct correlation between a standard mode and a custom mode this should be treated as just one mode (the standard mode) and emitted once.
+- The GCS can request this message using `MAV_CMD_REQUEST_MESSAGE`, specifying either "send all modes" or "send mode with this index".
+- The message will include a mode name that will be used to provide a string for the GCS to represent custom modes.
+  It would be empty for standard modes, for which name strings and translations may be hard coded into the GCS.
+- The flight should only emit each mode once.
+  If a mode is both custom and standard it should be emitted as a "standard mode" (this allows the GCS to list the "standard modes" and separately show the additional "custom modes").
+- The modes that are emitted depend on the current vehicle type.
+  For example, "takeoff" would not be emitted for a rover type, but would for a copter.
+
+A proposal for the message is provided below.
+
+```xml
+    <message id="435" name="AVAILABLE_MODES">
+      <description>Get information about a particular flight modes.
+        The message can be enumerated or requested for a particular mode using MAV_CMD_REQUEST_MESSAGE.
+        Specify 0 in param2 to request that the message is emitted for all available modes or the specific index for just one mode.
+        The modes must be available/settable for the current vehicle/frame type.
+        Each modes should only be emitted once (even if it is both standard and custom).
+      </description>
+      <field type="uint8_t" name="number_modes">The total number of available modes for the current vehicle type.</field>
+      <field type="uint8_t" name="mode_index">The current mode index within number_modes, indexed from 1.</field>
+      <field type="uint8_t" name="standard_mode" enum="MAV_STANDARD_MODE">Standard mode.</field>
+      <field type="uint8_t" name="base_mode" enum="MAV_MODE_FLAG" display="bitmask">System mode bitmap.</field>
+      <field type="uint32_t" name="custom_mode">A bitfield for use for autopilot-specific flags</field>
+      <field type="char[50]" name="mode_name">Name of custom mode, without null termination character. Should be omitted for standard modes.</field>
+    </message>
+```
+
+## Getting Current Active Mode
```

Comment:
@auturgy @LorenzMeier This section now suggests a separate message for getting the current mode, rather than using the `HEARTBEAT`. This was as I understand it discussed in https://github.com/mavlink/mavlink/wiki/20220119-Dev-Meeting.